store/acitreestore/tree.go

```diff
@@ -104,20 +242,26 @@ func (ts *TreeStore) Write(id string, key string, s *Store) (string, error) {
 		return "", errwrap.Wrap(errors.New("failed to sync tree store directory"), err)
 	}
 
+	// TODO(sgotti) this is wrong for various reasons:
```

Comment:
Thanks a lot for the xref to #2890, I didn't have the mental connection while reviewing. :-) If it makes, sense, I'd suggest to go ahead and open a separate issue for this specific problem.